src/Common/SharedAccessSignatureHelper.php

```diff
@@ -64,6 +64,156 @@ public function __construct($accountName, $accountKey)
     }
 
     /**
+     * Generates a shared access signature at the service level.
+     *
+     * This only supports version 2015-04-05 and later.
+     *
+     * @link https://msdn.microsoft.com/library/dn140255.aspx Constructing a service SAS.
+     *
+     * @param  string $signedPermissions  Signed permissions.
+     * @param  string $signedService      Service to generate the canonicalized resource.
+     *                                    It needs to be:
+     *                                    - b: for blob
+     *                                    - f: for file
+     *                                    - q: for queue
+     *                                    - t: for table
+     * @param  string $signedResource     Resource name to generate the canonicalized resource.
+     *                                    It can be:
+     *                                    - {cont}: for container
+     *                                    - {share}: for file shares
+     *                                    - {queue}: for queue
+     *                                    - {table}: for table
+     *                                    - {cont}/{blob}: for blobs
+     *                                    - {share}/{file}: for files
+     * @param  string $signedExpiry       Signed expiry date.
+     * @param  string $signedStart        Signed start date.
+     * @param  string $signedIdentifier   Signed identifier.
+     * @param  string $signedIP           Signed IP address.
+     * @param  string $signedProtocol     Signed protocol.
+     * @param  string $cacheControl       Cache-Control header (rscc).
+     * @param  string $contentDisposition Content-Disposition header (rscd).
+     * @param  string $contentEncoding    Content-Encoding header (rsce).
+     * @param  string $contentLanguage    Content-Language header (rscl).
+     * @param  string $contentType        Content-Type header (rsct).
+     *
+     * @return string                     Service Shared Access Signature.
+     */
+    public function generateServiceSharedAccessSignatureToken(
```

Comment:
Suggest to split this function to 4 functions and put them into the service proxies classes, e.g. BlobRestProxy. 

Several concerns:

1. Content-* headers are only for blob & file services only. 
2. For each services, the allowed permission are different. 
3. startpk, startrk, endpk, endrk only applicable for table service. 

Combining all the parameters in the same function will make users confused. :)